### PR TITLE
Kernel method to list all procedures

### PR DESCRIPTION
**Pull request by pontusmelke**

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/api/ProcedureRead.java [hunk 1]
```diff
@@ -33,6 +34,9 @@
     /** Fetch a procedure given its signature. */
     ProcedureSignature procedureGet( ProcedureSignature.ProcedureName name ) throws ProcedureException;
 
+    /** List all procedures */
+    Iterator<ProcedureSignature> proceduresGetAll();
```
Comment (lutovich) [inline community/kernel/src/main/java/org/neo4j/kernel/api/ProcedureRead.java]: Are values inside this iterator unique? Maybe use `Set` as the return type?

Comment (pontusmelke) [inline community/kernel/src/main/java/org/neo4j/kernel/api/ProcedureRead.java]: They are unique, trying to register the same signature multiple times will throw. I think having it return a `Set` makes sense.

File: community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/CallProcedureAcceptanceTest.scala [hunk 2]
```diff
@@ -42,9 +42,9 @@ class CallProcedureAcceptanceTest extends ExecutionEngineFunSuite {
     // Then
     result.toList should equal(
       List(
-        Map("label" -> "A"),
-        Map("label" -> "B"),
-        Map("label" -> "C")))
+        Map("labels" -> "A"),
```
Comment (lutovich) [inline community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/CallProcedureAcceptanceTest.scala]: Why key changed? If key can only point to a single string name of the label than I think it is better to keep it 'label'

**Comment by lutovich** [comment]
@pontusmelke needs a rebase

**Pull request merged**